src/shared/ask-password-api.c

```diff
@@ -68,7 +68,10 @@ static int lookup_key(const char *keyname, key_serial_t *ret) {
         assert(keyname);
         assert(ret);
 
-        serial = request_key("user", keyname, NULL, 0);
+        serial = keyctl(KEYCTL_SEARCH,
+                        KEY_SPEC_USER_KEYRING,
+                        (unsigned long) "user",
+                        (unsigned long) keyname, 0);
```

Comment:
Do we want the kernel to do a user callback? IMHO using KEYCTL_SEARCH would be the correct thing to do, no?

Anyway... request_key() does not work with current code. Let's see how things go if we link the keyrings.